File: Solutions/Java/2176.java
class Solution {
    public int countPairs(int[] nums, int k) {
        int n = nums.length;
        int count = 0;
        for (int i=0; i<n; i++) {
            for (int j=i+1; j<n; j++) {
                if (nums[i] == nums[j] && (i*j) % k == 0) {
                    count += 1;
                }
            }
        }
        return count;
    }
}

File: Solutions/Python/2176.py
class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        n = len(nums)
        count = 0
        for i in range(0, n):
            for j in range(i+1, n):
                if (nums[i] == nums[j]) and ((i*j) % k == 0):
                    count += 1
        return count
